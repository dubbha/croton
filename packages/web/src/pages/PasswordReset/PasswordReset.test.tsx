import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

describe('pages/PasswordReset', () => {
  const fn = jest.fn()

  it('should render successfully', () => {
    jest.isolateModules(() => {
      jest.doMock('react-redux', () => ({
        useStore: () => ({
          getState: () => ({}),
        }),
        useDispatch: () => fn,
      }));
  
      jest.doMock('components/PasswordResetForm', () => {
        const React = require('react');
        return {
          PasswordResetForm: ({ onSubmit }) => <button onClick={onSubmit} data-testid="submit" />,
        };
      });

      const { PasswordReset } = require('./PasswordReset');

      const { container } = render(<PasswordReset />, { wrapper: MemoryRouter });
      expect(container.firstChild).toMatchSnapshot();
    })
  })

  it('should dispatch on submit', () => {
    jest.isolateModules(() => {
      jest.doMock('react-redux', () => ({
        useStore: () => ({
          getState: () => ({}),
        }),
        useDispatch: () => fn,
      }));
  
      jest.doMock('components/PasswordResetForm', () => {
        const React = require('react');
        return {
          PasswordResetForm: ({ onSubmit }) => <button onClick={onSubmit} data-testid="submit" />,
        };
      });

      const { PasswordReset } = require('./PasswordReset');

      const { getByTestId } = render(<PasswordReset />, { wrapper: MemoryRouter });
      const submitButton = getByTestId('submit');

      fireEvent.click(submitButton);
      expect(fn).toBeCalled();
    })
  })
})