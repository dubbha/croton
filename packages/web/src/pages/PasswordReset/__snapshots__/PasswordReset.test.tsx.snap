// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/PasswordReset should render password reset form successfully 1`] = `
<div
  class="container"
>
  <div>
    Header
  </div>
  <button
    class="resetForm"
    data-testid="submitButton"
  />
  <div>
    Footer
  </div>
</div>
`;

exports[`pages/PasswordReset should render password update form successfully if token provided 1`] = `
<div
  class="container"
>
  <div>
    Header
  </div>
  <button
    class="updateForm"
    data-testid="submitButton"
  />
  <div>
    Footer
  </div>
</div>
`;
